### DISCUSSION TITLE
Clickhouse schema_migrations table doesn't support cluster wide replication

### DISCUSSION THREAD

**Discussion by oshankkumar** [amacneil/dbmate#356] [Feature Ideas]
My migrations look like this 
```sql
CREATE TABLE company_db.events ON CLUSTER 'company_cluster' (
    time DateTime,
    uid  Int64,
    type LowCardinality(String)
)
ENGINE = ReplicatedMergeTree('/clickhouse/tables/{cluster}/{shard}/table', '{replica}')
PARTITION BY toDate(time)
ORDER BY (uid);
```
After running the migration I'm expecting `schema_migrations` table  to be replicated across the cluster , So that I've a consistent schema migration information across cluster

[golang-migrate](https://github.com/golang-migrate/migrate/tree/master/database/clickhouse#readme) solved this issue by specifying `x-cluster-name` query parameter in DSN string.

**Comment by Flamefork** [comment]
Seems it wouldn't be hard to add such support in dbmate, just need some cluster to test against.
Is there a nice way to spin up a cluster on a local machine?

**Comment by oshankkumar** [comment]
You can easily spin up a cluster on local machine using docker compose. Please check this repo https://github.com/tetafro/clickhouse-cluster for reference.

**Comment by FatherCandle** [comment]
Hi guys, 
I have implemented the request with tests.
https://github.com/amacneil/dbmate/pull/432

You are more than welcome to review it and upvote if you think it looks mature enough to be merged :)
